File: ruby/ticket_item.rb
class TicketItem < GenericItem

  def update_quality
    quality_change =
      if @item.sell_in > 10
        1
      elsif @item.sell_in > 5
        2
      elsif @item.sell_in > 0
        3
      else
        -@item.quality
      end
    @quality_rating.modify(quality_change)
    @item.quality = @quality_rating.get
    --@item.sell_in
  end
end

File: Java/src/main/java/com/gildedrose/TicketItem.java
package com.gildedrose;

/**
 *
 * @author markd
 */
public class TicketItem extends GenericItem{
    
    public TicketItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }
    
    @Override
    public void updateQuality(){
        int qualityChange = 0;
        if(sellIn > 10){
            qualityChange = 1;
        }
        else if (sellIn > 5) {
            qualityChange = 2;
        }
        else if (sellIn > 0) {
            qualityChange = 3;
        }
        else{
            qualityChange = -quality;
        }
        qualityRating.modify(qualityChange);
        quality = qualityRating.get();
        sellIn--;
    }
    
}
